src/utils/scroll-lock.js

```diff
@@ -73,23 +74,30 @@ function advancedUnlock(targetElement) {
   document.removeEventListener('touchmove', preventDefault);
 }
 
+const isVerticalScroll = ({ scrollHeight, scrollWidth }) => scrollHeight > scrollWidth;
+
 /**
  * Handles the scrolling of the targetElement
  * @param {TouchEvent} event
  * @param {HTMLElement} targetElement
  * @return {boolean}
  */
-function handleScroll(event, targetElement) {
+function handleScroll(event, target) {
   const clientY = event.targetTouches[0].clientY - initialClientY;
-  // check if any parent has a scroll-lock disable, if not use the targetElement
-  const target = event.target.closest(`[${SCROLL_LOCK_DISABLE_ATTR}]`) || targetElement;
-  if (target.scrollTop === 0 && clientY > 0) {
-    // element is at the top of its scroll.
-    return preventDefault(event);
-  }
+  const clientX = event.targetTouches[0].clientX - initialClientX;
 
-  if (isTargetElementTotallyScrolled(target) && clientY < 0) {
-    // element is at the bottom of its scroll.
+  if (isVerticalScroll(target)) {
+    if (target.scrollTop === 0 && clientY > 0) {
+      // element is at the top of its scroll.
+      return preventDefault(event);
+    }
+
+    if (isTargetElementTotallyScrolled(target) && clientY < 0) {
+      // element is at the bottom of its scroll.
+      return preventDefault(event);
+    }
+  } else if (Math.abs(clientY) > Math.abs(clientX)) {
+    // prevent event if user tries to perform vertical scroll in an horizontal scrolling element
```

Comment:
I don't have idea about how WebKit really works but here I'm just comparing the absolute value of `clientY` (how much the user has moved to the Y axis since it started the movement) vs the absolute value of `clientX`(how much the user has moved to the X axis since it started the movement).

So if the user has moved more in the Y axis than to the X axis, it's a vertical scroll.